buffer_pool_for_uploading.go

```diff
@@ -0,0 +1,27 @@
+package goseaweedfs
+
+import "github.com/oxtoacart/bpool"
+
+// !! For user who wants to use buffer pool when uploading, please explicitly call SetBufferPoolForUploading(poolSize, BufferSize) !!
+
+var gBufferPool *bpool.SizedBufferPool
+
+////BufferPoolOptionsForUploading : will be used when uploading with multipart which will pre-allocate and reuse memory, and reduce memory usage significantly if we can estimate the file size we are uploading.
+//type BufferPoolOptionsForUploading struct {
+//	PoolSize   int // how many buffers in the pool, eg: 16
+//	BufferSize int // length of a buffer, eg: 32 * 1024 * 1024
+//}
+//
+//func NewBufferPoolOptionsForUploading(poolSize int, bufferSize int) *BufferPoolOptionsForUploading {
+//	return &BufferPoolOptionsForUploading{PoolSize: poolSize, BufferSize: bufferSize}
+//}
+
+//SetUploadOptions : will be used when uploading with multipart which will pre-allocate and reuse memory, and reduce memory usage significantly if we can estimate the size of uploading file.
+func SetBufferPoolForUploading(poolSize, bufferSize int) *bpool.SizedBufferPool {
```

Comment:
Along with above comment of building our own buffer pool. I think we could put predefined buffer-size into client's configuration instead of set it as global like this. 

We need one-time initialization instead of giving user the way to do it at runtime.